src/goto-instrument/contracts/instrument_spec_assigns.cpp

```diff
@@ -91,30 +94,119 @@ void instrument_spec_assignst::check_inclusion_assignment(
 
 void instrument_spec_assignst::track_static_locals(goto_programt &dest)
 {
-  auto call_graph =
-    call_grapht::create_from_root_function(functions, function_id, true)
-      .get_directed_graph();
+  covered_linest covered_lines;
+  const auto &found = functions.function_map.find(function_id);
+  INVARIANT(
+    found != functions.function_map.end(),
+    "the instrumented function must exist in the model");
+
+  const goto_programt &body = found->second.body;
+  traverse_instructions(
+    body.instructions.begin(), body.instructions.end(), covered_lines);
+
+  std::unordered_set<symbol_exprt, irep_hash> symbols;
+  collect_static_symbols(covered_lines, symbols);
 
-  for(const auto &sym_pair : st)
+  for(const auto &sym : symbols)
+    create_snapshot(create_car_from_stack_alloc(sym), dest);
+}
+
+size_t line_to_size_t(const irep_idt &line)
+{
+  PRECONDITION(!line.empty());
+  std::stringstream stream(id2string(line));
+  size_t res;
+  stream >> res;
+  return res;
+}
```

Comment:
You can (should) use one of the utility functions from `util/string2int.{h,cpp}` to take care of this for you.